tests/v3_api/test_secrets.py

```diff
@@ -0,0 +1,331 @@
+    keyvaluepair = {"testalldelete": "valuealldelete"}
+
+    secret = p_client.create_secret(name=name, namespaceId='NULL',
+                                    stringData=keyvaluepair)
+    print secret
+    assert secret['baseType'] == "secret"
+    secretdata = secret['data']
+
+    for k, v in secretdata.items():
+        key = k
+        vdecoded = base64.b64decode(v)
+
+    assert key == keyvaluepair.keys()[0]
+    assert vdecoded == keyvaluepair.values()[0]
+
+    print "Delete Secret"
+    p_client.delete(secret)
+
+    # Sleep to allow for the secret to be deleted
+    time.sleep(5)
+    print "Secret list after deleting one secret"
+    secretdict = p_client.list_secret()
+    print secretdict
+    print secretdict.get('data')
+    if len(secretdict.get('data')) > 0:
+        testdata = secretdict.get('data')
+        print "TESTDATA"
+        print testdata[0]['data']
+        if key in testdata[0]['data']:
+            assert False
+        else:
+            assert True
+    if len(secretdict.get('data')) == 0:
+        assert True
+
+    # Verify secret is deleted by "kubectl get secret" command
+    command = " get secret " + secret['name'] + " --namespace=" + ns.name
+    print "Command to obtain the secret"
+    print command
+    result = execute_kubectl_cmd(command, json_out=False, stderr=True)
+    print result
+
+    print "Verify that the secret does not exist " \
+          "and the error code returned is non zero "
+    if result != 0:
+        assert True
+
+
+def test_delete_secret_single_ns():
+
+    p_client = namespace["p_client"]
+    ns = namespace["ns"]
+    name = random_test_name("default")
+    keyvaluepair = {"testadelete": "valuedelete"}
+
+    secret = p_client.create_namespaced_secret(name=name,
+                                               namespaceId=ns['name'],
+                                               stringData=keyvaluepair)
+    print secret
+    assert secret['baseType'] == "namespacedSecret"
+    secretdata = secret['data']
+
+    for k, v in secretdata.items():
+        key = k
+        vdecoded = base64.b64decode(v)
+
+    assert key == keyvaluepair.keys()[0]
+    assert vdecoded == keyvaluepair.values()[0]
+
+    print "Delete Secret"
+    p_client.delete(secret)
+
+    # Sleep to allow for the secret to be deleted
+    time.sleep(5)
+    print "Secret list after deleting one secret"
+    secretdict = p_client.list_namespaced_secret()
+    print secretdict
+    print secretdict.get('data')
+    if len(secretdict.get('data')) == 0:
+        assert True
+
+    # Verify secret is deleted by "kubectl get secret" command
+    command = " get secret " + secret['name'] + " --namespace=" + ns.name
+    print "Command to obtain the secret"
+    print command
+    result = execute_kubectl_cmd(command, json_out=False, stderr=True)
+    print result
+
+    print "Verify that the secret does not exist and the error " \
+          "code returned is non zero "
+    if result != 0:
+        assert True
```

Comment:
can we have a common function to check for the delete secret that can be used by both test_delete_secret_all_ns and test_delete_secret_single_ns ?